source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -829,9 +829,10 @@ def fill_journeys(self, request_type, api_request, instance):
         nb_try = 0
         nb_qualified_journeys = 0
 
+        max_journeys_calls = app.config.get('MAX_JOURNEYS_CALLS', 20)
         while request is not None and \
-                ((nb_qualified_journeys < min_nb_journeys and nb_try < min_nb_journeys) \
-                or nb_try < min_journeys_calls):
+                ((nb_qualified_journeys < min_nb_journeys and nb_try < min(min_nb_journeys, max_journeys_calls))\
```

Comment:
The condition is pretty convoluted already, would be much nicer to extract the `min(min_nb_journeys, max_journeys_calls))` in a separate variable above the loop.

Reply:
or put all in a function